tests/steamship_tests/plugin/unit/trainable/test_model_checkpoints.py

```diff
@@ -42,20 +42,41 @@ def test_model_checkpoint_save_load():
     checkpoint_1 = ModelCheckpoint(client=client, handle="epoch1", plugin_instance_id="0000")
     with open(checkpoint_1.folder_path_on_disk() / "params.json", "w") as f:
         f.write("HI THERE")
-    checkpoint_1.upload_model_bundle()
-
-    # Now let's download the checkpoint labeled "epoch1" and test that it is equal
-    checkpoint_downloaded = ModelCheckpoint(
-        client=client, handle="epoch1", plugin_instance_id="0000"
-    )
-    checkpoint_downloaded.download_model_bundle()
+    update_version = checkpoint_1.upload_model_bundle()
+
+    version_mismatch = True
+    attempts = 5
+    while version_mismatch and attempts > 0:
+        # Now let's download the checkpoint labeled "epoch1" and test that it is equal
+        checkpoint_downloaded = ModelCheckpoint(
+            client=client, handle="epoch1", plugin_instance_id="0000"
+        )
+        _, model_version = checkpoint_downloaded.download_model_bundle()
+        if update_version != model_version:
+            attempts -= 1
+            update_version = checkpoint_1.upload_model_bundle()
+        else:
+            version_mismatch = False
+
+    assert attempts > 0
     _test_folders_equal(
```

Comment:
i wonder if I can just use `_test_folders_equal` instead of hashing each upload/download.